lms/static/js/inline_analytics.js

```diff
@@ -94,66 +97,55 @@ window.InlineAnalytics = (function() {
 
         // Build the array of choice texts
         var choiceText = getChoiceTexts(partId);
+        
+    	// Generate choice name array
+    	var choiceNameArray = convertNameToArray(choiceNameString);
 
         if (result) {
-            // Loop through results and construct row array
-            // Rows returned from the analytics API are in "option" order so we use this
-            // to determine if any rows are "missing" from the data.
-            var arrayLength = result.length;
+            // Build the array of value_id's
+            var valueIdArray = constructValueIdArray(result);
+        	
+            // Loop through choiceNameArray and construct row array.
+            // We can determine if any rows are "missing" from the api data
+            // since the choiceNameArray is a list of all possible rows.
+            var arrayLength = choiceNameArray.length;
```

Comment:
Right, the assignment will still take place here, but the `var arrayLength;` will move up above with the other declarations.